docs/docs/creating-and-modifying-pages.md

```diff
@@ -92,13 +92,103 @@ exports.createPages = async ({ graphql, actions, reporter }) => {
+
+```javascript:tittle=gatsby-node.js
+
+exports.createPages = async ({ graphql, actions }) => {
+  const { createPage } = actions;
+  const queryResults = await graphql(`
+    query AllProducts {
+      AllProducts {
+        nodes {
+          id
+          name
+          price
+          description
+        }
+      }
+    }
+  `);
+
+  const productTemplate = path.resolve(`src/templates/product.js`);
+  queryResults.data.allProducts.nodes.forEach(node => {
+    createPage({
+      path: `/products/${node.id}`,
+      component: productTemplate,
+      context: {
+        // This time the entire product is passed down as context
+        product
+      }
+    });
+  });
+};
+};
+```
+
+> You are now requesting all the data you need in a single query (this requires server-side support to fetch many products in a single database query).
+
+> As long as you can pass this data down to the template component via `pageContext`, there is no need for the template to make a GraphQL query at all.
+
+Your template file would look like this:
+
+```javascript:tittle=src/templates/product.js
+function Product({ pageContext }) {
+  return (
+    <div>
+      Name: {pageContext.name}
+      Price: {pageContext.price}
+      Description: {pageContext.description}
+    </div>
+  )
+}
+```
+
+### Pefromance implecations
+
+Using the PageContext props in the template component can come with its performance advantages, of getting in all the data you need at build time; from the createPages API. This prevents having a graphql query in the template component.
+
+It does come with the advantage of querying your data from one place after declaring the `context` parameter.
+
+However, it doesn’t give you the opportunity to know what exactly you are querying for in the template and if any changes occur in the component query structure in gatsby-node.js. Hot reload is taken off the table and the site needs to be rebuilt for changes to reflect.
+
+Gatsby stores page metadata (including context) in a redux store (which also means that it stores its memory). For large enough sites (either number of pages and/or amount of data that is being passed via page context) this will cause problems. This might be out of memory crashes if it's too much data or degraded performance.
+
+> If there is memory pressure, Node.js will try to garbage collect more often, which is a known performance issue.
+
+Page query results, are not stored in memory permanently and are being saved to disk immediately after running the query.
+
+We recommend passing "ids" or "slugs" and making full queries in the page template query to avoid this.
+
+### Incremental builds trade-off of this method
+
+Another disadvantage of this method is that because your site has to be rebuilt every time you make a change you will not be able to take advantage of incremental builds.
```

Comment:
```suggestion
Another disadvantage of querying all of your data in `gatsby-node.js` is that your site has to be rebuilt every time you make a change, so you will not be able to take advantage of incremental builds.
```

Restating what the method was might make sense here, the first time I read it I thought it was referring to the method of passing in an id and then querying in the template since that's the last thing I read the section before.